### PR TITLE
[Pulumi] updates to pages under `/pulumi`

### PR DESCRIPTION
**Pull request by desteves**
### Summary

The following all apply to existing`/pulumi` content:

* The overview page now links to the official Pulumi Docs
* Renames installation page to "Get started" for clarity
* Update weights for menu item reordering
* Minor Markdown edits such as removing trailing whitespaces
* Extensive updates to the existing "Add a site" tutorial including:
  * All code now uses a language picker for all supported Pulumi programming languages
  * Updates storing the settings in Pulumi ESC as a best practice
  * All code snippets have been tested locally across all programming languages via `pulumi preview` commands.

### Documentation checklist

- [ x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [x] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [x] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: content/pulumi/tutorial/add-site.md [hunk 1]
```diff
@@ -64,61 +55,245 @@ $ pulumi login
...
+{{</Aside>}}
 
-A Pulumi program is code written in a [supported programming language](https://www.pulumi.com/docs/languages-sdks/) that defines infrastructure resources. We use TypeScript.
+To create a program, select your language of choice and run the `pulumi` command:
 
-{{</Aside>}}
+{{<tabs labels="js | ts | py | Go | Java | .NET | YAML">}}
+{{<tab label="js" default="true">}}
+
+```bash
```
Comment (kodster28) [inline content/pulumi/tutorial/add-site.md]:
```suggestion
```sh
```

Comment (kodster28) [reply content/pulumi/tutorial/add-site.md]: I might switch these all to our `sh` language block, which is really just bash but set up to only grab the command.

That'll let you copy/paste just the `pulumi new ...` and ignore the comment.

File: content/pulumi/tutorial/add-site.md [hunk 2]
```diff
@@ -64,61 +55,245 @@ $ pulumi login
...
 - Your Cloudflare [account ID](/fundamentals/setup/find-account-and-zone-ids/).
-- A registered domain. For instance, `example.com`
 - A valid Cloudflare API [token](/fundamentals/api/get-started/create-token/).
+- A domain. For instance, `example.com`.
+
+{{<Aside type="note">}}
+A Pulumi [ESC Environment](https://www.pulumi.com/docs/esc/environments/) is a YAML file containing configurations and secrets that pertain to your application and infrastructure. These can be accessed in several ways, including a Pulumi program. All ESC Environments reside in your Pulumi Cloud account.
+{{</Aside>}}
+
+```bash
```
Comment (kodster28) [inline content/pulumi/tutorial/add-site.md]: Unless you want the other comments to be selectable, might be worth changing to `sh` as well here.

**Review by kodster28 — APPROVED** [review]
Pushed up a commit to fix a few weird things in tabs b/c our tab component isn't super intuitive / well-built.

Added a few small comments, mostly just about code block formatting.

Let me know when this is fine to merge and happy to do that.

File: content/pulumi/tutorial/add-site.md (hunk 1 above)
Comment (desteves) [reply content/pulumi/tutorial/add-site.md]: Ahh, gotcha; thanks for letting me know. I'll keep this in mind!

**Comment by desteves** [comment]
Thanks, @kodster28. I updated the code blocks, so it should be good to go from my end.

**Pull request merged**